### PR TITLE
chore(deps): upgrade deps to latest version

### PR DESCRIPTION
**Pull request by herberthenrique**
Addresses #issue-number.

## Purpose

Upgrade dependencies to latest versions

## Approach and changes

- Upgrade foundry inside the project
- Upgrade other libs that are inside the generated project

## Definition of done

- [x] Development completed
- [x] Reviewers assigned
- [x] Unit and integration tests

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -1,16 +1,31 @@
...
+        },
+      },
+      {
+        files: ['files/**/*.spec.js'],
+        env: {
+          jest: true,
+        },
+      },
+    ],
+  },
```
Comment (connor-baer) [inline .eslintrc.js]: None of this should be required anymore if you're using Foundry.

File: .eslintrc.js [hunk 2]
```diff
@@ -1,16 +1,31 @@
...
-      }
-    }
-  ]
-};
-
-module.exports = overwritePresets(baseConfig, customConfig);
+module.exports = require('@sumup/foundry/eslint')(
+  {
+    language: 'JavaScript',
+    environments: ['Node'],
```
Comment (connor-baer) [inline .eslintrc.js]:
```suggestion
    environments: ['Node', 'Browser'],
```

File: package.json [hunk 3]
```diff
@@ -27,10 +27,10 @@
     "chalk": "^2.4.2",
     "esm": "^3.2.25",
     "listr": "^0.14.3",
-    "listr-verbose-renderer": "^0.5.0"
+    "listr-verbose-renderer": "^0.5.0",
+    "@sumup/foundry": "^2.2.0"
```
Comment (connor-baer) [inline package.json:31]: Why did you move Foundry to dependencies?

File: .eslintrc.js (hunk 1 above)
Comment (herberthenrique) [reply .eslintrc.js]: If I don't make this happen, lint was complaning 🤷‍♂️ 
the override in `files/**/*.js` was here before

File: package.json (hunk 3 above)
Comment (herberthenrique) [reply package.json:31]: Because lint was complaining

File: .eslintrc.js (hunk 1 above)
Comment (connor-baer) [reply .eslintrc.js]: That's what I mean, those overrides are obsolete now 😉

File: package.json (hunk 3 above)
Comment (connor-baer) [reply package.json:31]: Where was it complaining? What was the error message?

**Review by connor-baer — APPROVED** [review]
Not entirely convinced those custom rules are necessary, but I plan to rewrite the entire library for v2 anyway 🤷‍♂️

File: package.json (hunk 3 above)
Comment (herberthenrique) [reply package.json:31]: Oh, sorry.
Inside the foundry configuration file, was showing an error about foundry being on devDependencies

File: .eslintrc.js (hunk 1 above)
Comment (herberthenrique) [reply .eslintrc.js]: But without the overrides, the lint is complaining about the jest globals not being defined 🤔

File: package.json (hunk 3 above)
Comment (connor-baer) [reply package.json:31]: Those config files should be ignored by Eslint in Foundry v2.2. Anyway 🤷‍♂️

File: .eslintrc.js (hunk 1 above)
Comment (connor-baer) [reply .eslintrc.js]: You're missing `frameworks: ['Jest']` in the config

Comment (connor-baer) [reply .eslintrc.js]: https://github.com/sumup-oss/foundry#eslint

Comment (herberthenrique) [reply .eslintrc.js]: AH, got it
